### PR TITLE
Add recursive dossier creation for dossiertemplates

### PR DESCRIPTION
**Pull request by elioschmutz**
Dieser PR erstellt rekursiv Dossiers und Dokumente anhand des Dossiertemplates.
Zusätzlich fügt er zwei Dossiertemplates als Examplecontent hinzu.

screen3

see #2143

### CODE REVIEW

**Comment by elioschmutz** [comment]
@phgross ready to review...

File: opengever/base/command.py [hunk 1]
```diff
@@ -10,7 +10,36 @@
...
+
+        self.finish_creation(obj)
+
+        obj.reindexObject()
+        return obj
+
+    def finish_creation(self, content):
+        """Trigger rest of initialization."""
+
+        notify(ObjectModifiedEvent(content))
```
Comment (phgross) [inline opengever/base/command.py]: Please doublecheck that the event is not fired multiple times.

File: opengever/dossier/command.py [hunk 2]
```diff
@@ -19,3 +21,24 @@ def finish_creation(self, content):
 
         DocPropertyWriter(content, recipient_data=self.recipient_data).initialize()
         super(CreateDocumentFromTemplateCommand, self).finish_creation(content)
+
+
+class CreateDossierFromTemplateCommand(BaseObjectCreatorCommand):
+    """Store a copy of the dossiertemplate with all its attributes.
```
Comment (phgross) [inline opengever/dossier/command.py]: Store a copy 😏

File: opengever/dossier/command.py [hunk 3]
```diff
@@ -19,3 +21,24 @@ def finish_creation(self, content):
...
+
+    def __init__(self, context, template):
+        super(CreateDossierFromTemplateCommand, self).__init__(
+            context, template.title, **self._get_additional_attributes(template))
+
+    def _get_additional_attributes(self, template):
+        """Extracts the template attributes to use it as additional attributes.
+        """
+        data = get_persisted_values_for_obj(template)
+        del data['title']  # the titel is already given as a parameter
```
Comment (phgross) [inline opengever/dossier/command.py]: titel -> title

Comment (phgross) [inline opengever/dossier/command.py]: why?

File: opengever/dossier/dossiertemplate/form.py [hunk 4]
```diff
@@ -196,16 +199,29 @@ def handleAdd(self, action):
 
                 obj = self.createAndAdd(data)
                 if obj is not None:
-                    # mark only as finished if we get the new object
-                    self._finishedAdd = True
+                    obj = api.content.find(context=self.context, depth=1, id=obj.getId())[0].getObject()
```
Comment (phgross) [inline opengever/dossier/dossiertemplate/form.py]: pyflakes

Comment (phgross) [inline opengever/dossier/dossiertemplate/form.py]: obj = self.context.get(obj.getId()) ? or what's the reason for making a catalog query here?

File: opengever/dossier/tests/test_dossier_template.py [hunk 5]
```diff
@@ -322,3 +323,81 @@ def test_redirects_to_first_step_if_the_user_skips_the_first_wizard_step(self, b
         self.assertEqual(
             '{}/dossier_with_template'.format(self.leaf_node.absolute_url()),
             browser.url)
+
+    @browsing
+    def test_add_recursive_documents_and_subdossiers(self, browser):
+        templatedossier = create(Builder('templatedossier'))
+
+        root = create(Builder('repository_root'))
```
Comment (phgross) [inline opengever/dossier/tests/test_dossier_template.py]: there is already content defined in testcase setUp

File: opengever/dossier/tests/test_dossier_template.py [hunk 6]
```diff
@@ -322,3 +323,81 @@ def test_redirects_to_first_step_if_the_user_skips_the_first_wizard_step(self, b
         self.assertEqual(
             '{}/dossier_with_template'.format(self.leaf_node.absolute_url()),
             browser.url)
+
+    @browsing
+    def test_add_recursive_documents_and_subdossiers(self, browser):
```
Comment (phgross) [inline opengever/dossier/tests/test_dossier_template.py]: I would simplify this testcase but add separate testcase wich checks the document creation and one which checks the subdossier creation, including default values, correct catalog indexed etc.
 - test_adds_content_recursively
  - 1 subdossier with 1 document
  - 1 subsub with 1 document

**Comment by elioschmutz** [comment]
@phgross please take another look...

**Pull request merged**